File: PYTHON/PROGRAM/Cpu.py
from Komponen import Komponen

class Cpu(Komponen):
    #konstruktor
    def __init__(self, merk = "", nama = "", jumlahCore = 0, kecepatanGHz = 0):
        super().__init__(merk, nama)
        self.__jumlahCore = jumlahCore
        self.__kecepatanGHz = kecepatanGHz

    #getter dan setter
    def getJumlahCore(self):
        return self.__jumlahCore
    
    def setJumlahCore(self, jumlahCore):
        self.__jumlahCore = jumlahCore

    def getKecepatanGHz(self):
        return self.__kecepatanGHz
    
    def setKecepatanGHz(self, kecepatanGHz):
        self.__kecepatanGHz = kecepatanGHz

File: CPP/PROGRAM/Cpu.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Cpu : public Komponen{
    //atribut
    private:
        int jumlahCore;
        float kecepatanGHz;

    public:
        //konstruktor
        Cpu(){
            this->jumlahCore = 0;
            this->kecepatanGHz = 0;
        }

        Cpu(int jumlahCore, float kecepatanGHz, string merk, string nama) : Komponen(merk, nama){
            this->jumlahCore = jumlahCore;
            this->kecepatanGHz = kecepatanGHz;
        }

        //getter dan setter
        void setJumlahCore(int jumlahCore){
            this->jumlahCore = jumlahCore;
        }

        void setKecepatanGHz(float kecepatanGHz){
            this->kecepatanGHz = kecepatanGHz;
        }

        int getJumlahCore(){
            return this->jumlahCore;
        }

        float getKecepatanGHz(){
            return this->kecepatanGHz;
        }

        ~Cpu(){

        }
};
